src/main/java/nextstep/optional/User.java

```diff
@@ -33,7 +35,10 @@ public static boolean ageIsInRange1(User user) {
     }
 
     public static boolean ageIsInRange2(User user) {
-        return false;
+        Optional<User> userOptional = Optional.ofNullable(user);
+
+        return userOptional.map(User::getAge).filter(age -> age>=30).filter(age -> age<=45).isPresent();
```

Comment:
각각의 `.filter()`에서 처리하는 로직은 하나로 합치는 게 더 좋을 것 같습니다.
```suggestion
    return Optional.ofNullable(user)
            .map(User::getAge)
            .filter(age -> age >= 30 && age <= 45)
            .isPresent();
```